src/main/java/org/apache/commons/io/IOUtils.java

```diff
@@ -752,19 +777,46 @@ public static boolean contentEquals(final Reader input1, final Reader input2)
         if (input1 == null ^ input2 == null) {
             return false;
         }
-        final BufferedReader bufferedInput1 = toBufferedReader(input1);
-        final BufferedReader bufferedInput2 = toBufferedReader(input2);
 
-        int ch = bufferedInput1.read();
-        while (EOF != ch) {
-            final int ch2 = bufferedInput2.read();
-            if (ch != ch2) {
-                return false;
+        char[] charArray1 = new char[CONTENT_EQUALS_CHAR_ARRAY_BUFFER_SIZE];
+        char[] charArray2 = new char[CONTENT_EQUALS_CHAR_ARRAY_BUFFER_SIZE];
+        int nowPos1;
+        int nowPos2;
+        int nowRead1;
+        int nowRead2;
+        while (true) {
+            nowPos1 = 0;
+            nowPos2 = 0;
+            for (int nowCheck = 0; nowCheck < CONTENT_EQUALS_CHAR_ARRAY_BUFFER_SIZE; nowCheck++) {
+                if (nowPos1 == nowCheck) {
+                    do {
+                        nowRead1 = input1.read(charArray1, nowPos1, CONTENT_EQUALS_CHAR_ARRAY_BUFFER_SIZE - nowPos1);
+                    } while (nowRead1 == 0);
+                    if (nowRead1 == -1) {
+                        return nowPos2 == nowCheck && input2.read() == -1;
+                    }
+                    nowPos1 += nowRead1;
+                }
+                if (nowPos2 == nowCheck) {
+                    do {
+                        nowRead2 = input2.read(charArray2, nowPos2, CONTENT_EQUALS_CHAR_ARRAY_BUFFER_SIZE - nowPos2);
+                    } while (nowRead2 == 0);
+                    if (nowRead2 == -1) {
+                        return nowPos1 == nowCheck && input1.read() == -1;
+                    }
+                    nowPos2 += nowRead2;
+                }
+                if (charArray1[nowCheck] != charArray2[nowCheck]) {
+                    return false;
+                }
             }
-            ch = bufferedInput1.read();
         }
+    }
 
-        return bufferedInput2.read() == EOF;
+    private enum LastState {
+        r,
```

Comment:
Can you please use UPPERCASE identifiers?
Also, IMHO it is better to give meaningful names or add a minimal explanation

Reply:
> Can you please use UPPERCASE identifiers?
Also, IMHO it is better to give meaningful names or add a minimal explanation

@eolivelli
OK, will do.